packages/assistant/src/RPA/Assistant/library.py

```diff
@@ -0,0 +1,991 @@
+            END
+        """
+        self._client.add_element(
+            name=str(name), element=Checkbox(label=str(label), value=bool(default))
+        )
+
+    @keyword("Add submit buttons", tags=["input"])
+    def add_submit_buttons(
+        self,
+        buttons: Options,
+        default: Optional[str] = None,
+    ) -> None:
+        """Add custom submit buttons
+
+        :param buttons: Submit button options
+        :param default: The primary button
+
+        The dialog automatically creates a button for closing itself.
+        If there are no input fields, the button will say "Close".
+        If there are one or more input fields, the button will say "Submit".
+
+        If the submit button should have a custom label or there should be
+        multiple options to choose from  when submitting, this keyword can
+        be used to replace the automatically generated ones.
+
+        The result field will always be called ``submit`` and will contain
+        the pressed button text as a value.
+
+        If one of the custom ``options`` should be the preferred option,
+        the ``default`` argument controls which one is highlighted with
+        an accent color.
+
+        Example:
+
+        .. code-block:: robotframework
+
+            Add icon      Warning
+            Add heading   Delete user ${username}?
+            Add submit buttons    buttons=No,Yes    default=Yes
+            ${result}=    Run dialog
+            IF   $result.submit == "Yes"
+                Delete user    ${username}
+            END
+        """
+        buttons, default = to_options(buttons, default)
+        for button in buttons:
+            self._add_closing_button(button)
+
+    @keyword("Run dialog", tags=["dialog"])
+    def run_dialog(
+        self,
+        timeout: int = 180,
+        title: str = "Dialog",
+        height: Union[int, str] = "AUTO",
+        width: int = 480,
+        on_top: bool = False,
+    ) -> Result:
+        """Create a dialog from all the defined elements and block
+        until the user has handled it.
+
+        :param timeout: Time to wait for dialog to complete, in seconds
+        :param title:  Title of dialog
+        :param height: Height of dialog (in pixels or 'AUTO')
+        :param width:  Width of dialog (in pixels)
+        :param on_top: Show dialog always on top of other windows
+
+
+        Returns a result object with all input values.
+
+        Example:
+
+        .. code-block:: robotframework
+
+            Add heading     Please enter your username
+            Add text input  name=username
+            ${result}=      Run dialog
+            Log    The username is: ${result.username}
+        """
+
+        # FIXME: support timeout
+
+        self._client.display_flet_window(title, height, width, on_top)
+        return self._client.results
+
+    @keyword("Ask User", tags=["dialog"])
+    def ask_user(self, timeout: int = 180, **options: Any) -> Result:
```

Comment:
This keyword and what it does confuses me a bit. Or rather, I understand from looking at the code, but reading the documentation I'd be a bit baffled. Maybe rethink if this is required, and if it's a very useful shorthand think about integrating it somehow into the normal dialog configuration/running.

Reply:
It should be the default way of using Assistant dialogs (instead of run dialog) for simple cases, like forms. Agreed that the documentation could be better, but don't think that's the biggest priority right now, so maybe we can discuss about the docs later on. And yes, this keyword was a requirement